src/components/autocomplete-select/AutocompleteSelect.js

```diff
@@ -1,20 +1,36 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import Select from 'react-select';
-
 import styles from './AutocompleteSelect.css';
 
 const hideMenuWithNoOptions = () => 'No options available';
 
-const AutocompleteSelect = ({ className, ...rest }) => (
-    <Select
-        { ...rest }
-        className={ classNames(styles.container, className) }
-        classNamePrefix="autocomplete" />
-);
+class AutocompleteSelect extends Component {
+    render() {
+        const { className, ...rest } = this.props;
+
+        return (
+            <Select
+                { ...rest }
+                onChange={ this.handleSelectChange }
+                value={ this.getValueFromOptions() }
+                className={ classNames(styles.container, className) }
+                classNamePrefix="autocomplete" />
+        );
+    }
+
+    getValueFromOptions() {
```

Comment:
It would be nice for this function to be memoized, either by using memoize-one package or keeping this component functional and use useMemo and useCallback

Reply:
I was actually searching about it 😄